src/main/java/org/broadinstitute/hellbender/tools/copynumber/PreprocessIntervals.java

```diff
@@ -0,0 +1,128 @@
+                + "The length of the padding regions at both sides of the intervals (-P) and the length of the bins (-BL) can be given as command line arguments. ",
+        oneLineSummary = "Preprocess intervals given by the user and then create bins that cover them.",
+        programGroup = CopyNumberProgramGroup.class
+)
+@DocumentedFeature
+public final class PreprocessIntervals extends GATKTool {
+    public static final String LENGTH_OF_BINS_SHORT_NAME = "BL";
+    public static final String LENGTH_OF_BINS_LONG_NAME = "binLength";
+
+    public static final String PADDING_SHORT_NAME = "P";
+    public static final String PADDING_LONG_NAME = "padding";
+
+    @Argument(
+            doc = "Length (in bp) of the bins.",
+            fullName = LENGTH_OF_BINS_LONG_NAME,
+            shortName = LENGTH_OF_BINS_SHORT_NAME,
+            optional = true,
+            minValue = 1
+    )
+    private int binLength = 1000;
+
+    @Argument(
+            doc = "Length (in bp) of the padding regions on each side of the intervals.",
+            fullName = PADDING_LONG_NAME,
+            shortName = PADDING_SHORT_NAME,
+            optional = true,
+            minValue = 0
+    )
+    private int padding = 0;
+
+    @Argument(
+            doc = "Output file.",
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME
+    )
+    private File outputFile;
+
+    @Override
+    public void onTraversalStart() {
+        createBins();
+    }
+
+    private void createBins() {
+        final SAMSequenceDictionary sequenceDictionary = getBestAvailableSequenceDictionary();
+
+        // if the user didn't add any intervals, we assume that they wanted to do whole genome sequencing
+        final List<SimpleInterval> inputIntervals = hasIntervals() ? intervalArgumentCollection.getIntervals(sequenceDictionary)
+               : IntervalUtils.getAllIntervalsForReference(sequenceDictionary);
+        final IntervalList preprocessedIntervalList = padAndMergeIntervals(inputIntervals, padding, sequenceDictionary);
+        final IntervalList bins = generateBins(preprocessedIntervalList, padding, binLength, sequenceDictionary);
+
+        bins.write(outputFile);
+    }
+
+    private static IntervalList padAndMergeIntervals(final List<SimpleInterval> inputIntervals, final int padding, final SAMSequenceDictionary sequenceDictionary) {
+        final IntervalList inputIntervalList = new IntervalList(sequenceDictionary);
+        inputIntervals.stream().map(si -> new Interval(si.getContig(), si.getStart(), si.getEnd())).forEach(inputIntervalList::add);
+
+        final IntervalList uniquedIntervalList = inputIntervalList.uniqued();
+        final IntervalList paddedIntervalList = uniquedIntervalList.padded(padding, padding);
+        final IntervalList mergedIntervalList = IntervalList.intersection(paddedIntervalList, paddedIntervalList);
```

Comment:
As you can see from the highlighting in IntelliJ, you can inline the return statement if you like (`return IntervalList.intersection(paddedIntervalList, paddedIntervalList);`).  Up to you.

Reply:
I'd rather keep it like this for readability.